File: rb/features.rb
# Digimon SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DigimonFeatures
  def self.make_feature(name)
    case name
    when "base"
      DigimonBaseFeature.new
    when "ratelimit"
      DigimonRatelimitFeature.new
    when "retry"
      DigimonRetryFeature.new
    when "test"
      DigimonTestFeature.new
    when "timeout"
      DigimonTimeoutFeature.new
    else
      DigimonBaseFeature.new
    end
  end
end

File: py/features.py
# Digimon SDK feature factory

from feature.base_feature import DigimonBaseFeature
from feature.test_feature import DigimonTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DigimonBaseFeature(),
        "test": lambda: DigimonTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
